packages/@react-aria/radio/src/useRadio.ts

```diff
@@ -80,22 +81,25 @@ export function useRadio(props: AriaRadioProps, state: RadioGroupState, ref: Ref
     tabIndex = undefined;
   }
 
+  let {name, descriptionId, errorMessageId, validationBehavior} = radioGroupData.get(state)!;
```

Comment:
This seems to break when the `useRadio` and `useRadioGroup` hook are used in separate components.

```tsx
<RadioGroup>
  <RadioGroupItem />
</RadioGroup>
```

I think this is because in this setup the item is rendered first and then the parent is rendered. This worked before because the undefined values weren't being destructured.

Reply:
This is actually a common misconception. Parents are "rendered" first, but children's effects run first. See https://stackoverflow.com/questions/53781632/whats-useeffect-execution-order-and-its-internal-clean-up-logic-when-requestani for some examples.

Can you reproduce this in a codesandbox? Because we have an example in our docs where the hooks are in two different components and it looks like what you've outlined, but it's working for us https://react-spectrum.adobe.com/react-aria/useRadioGroup.html#example

Are you sure you've passed the correct state object from radio group to each radio?

Reply:
hey, thanks for the quick response!

You are right, I wasn't passing the correct state object. In addition to the `state` from `useRadioGroupState` I was passing extra fields through the context and then destructuring them when accessing the context. So this wasn't working for me prior either, it just wasn't throwing an error haha. After updating to use the same state object throughout it's working 👍

Here's a [codesandbox](https://codesandbox.io/p/sandbox/strange-dream-6j9tmg?file=%2Fsrc%2FApp.tsx%3A38%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clpa6ezde00072e67ka8plvp6%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clpa6ezde00032e67s8sv0qhm%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clpa6ezde00042e67rbexrpf1%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clpa6ezde00062e67pfahgyw0%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B60%252C40%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clpa6ezde00032e67s8sv0qhm%2522%253A%257B%2522id%2522%253A%2522clpa6ezde00032e67s8sv0qhm%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpa6ordo00032e67l0mumx5y%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A38%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A38%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clpa6ordo00032e67l0mumx5y%2522%257D%252C%2522clpa6ezde00062e67pfahgyw0%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clpa6ezde00052e67ouba1rfy%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%2522%257D%255D%252C%2522id%2522%253A%2522clpa6ezde00062e67pfahgyw0%2522%252C%2522activeTabId%2522%253A%2522clpa6ezde00052e67ouba1rfy%2522%257D%252C%2522clpa6ezde00042e67rbexrpf1%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clpa6ezde00042e67rbexrpf1%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D) showing the error I was seeing. 

Thanks again for the quick response!!